crates/proto/src/xfer/dns_multiplexer.rs

```diff
@@ -203,7 +205,7 @@ where
         // drop all the canceled requests
         for (id, error) in canceled {
             if let Some(active_request) = self.active_requests.remove(&id) {
-                if active_request.responses.is_empty() {
+                if active_request.response.is_none() {
```

Comment:
Does this distinction still make sense?

Reply:
Not sure exactly what you mean, but I think I could flatten the take into the get and remove the second check?

Reply:
I meant the distinction between what it actually means for `active_request.response.is_none()` to be true vs false and if those should be treated differently when we're conceptually returning a stream of responses.

Reply:
Fair enough, I didn't have that distinction as clearly in mind.

Reply:
Ah. Yes. It’s possible that we can get rid of the stored optional message all together, and just put it onto the channel immediately, but I need to look more thoroughly at the code.